docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,793 @@
+With the basic framework for the project established, let's walk through how to model our coffee service.
+
+As discussed above, the coffee service should:
+
+* provide a menu of coffees
+* provide the ability to order a coffee
+* provide the ability to check the status of an order
+
+------------------
+Adding a Service
+------------------
+The service shape is the entry-point of our API, and is where we define the operations that our service exposes to a
+consumer. With that information in mind, let's define the initial service shape without any operations:
+
+.. code-block:: smithy
+    :caption: ``main.smithy``
+
+    $version: "2.0"
+
+    namespace com.example
+
+    use aws.protocols#restJson1
+
+    /// Allows users to retrieve a menu, create a coffee order, and
+    /// and to view the status of their orders
+    @title("Coffee Shop Service")
+    @restJson1
+    service CoffeeShop {
+        version: "2024-04-04"
+    }
+
+We apply the ``@restJson1`` protocol trait to the service to indicate that this service supports the
+:doc:`../aws/protocols/aws-restjson1-protocol` created and used extensively by AWS.
+Put simply, protocols define the rules and conventions for how data is serialized and deserialized when communicating
+between client and server. Protocols are complex and need their own topic, so we will not discuss them at length in
+this tutorial.
+
+---------------
+Defining Shapes
+---------------
+Let's create basic representations of our data in Smithy. We can further refine our data model using traits.
+Open the file titled ``coffee.smithy``. We will use it to write our definitions of coffee-related structures:
+
+.. code-block:: smithy
+    :caption: ``coffee.smithy``
+
+    $version: "2.0"
+
+    namespace com.example
+
+    /// A enum describing the types of coffees available
+    enum CoffeeType {
+        DRIP
+        POUR_OVER
+        LATTE
+        ESPRESSO
+    }
+
+    /// A structure which defines a coffee item which can be ordered
+    structure CoffeeItem {
+        @required
+        type: CoffeeType
+
+        @required
+        description: String
+    }
+
+    /// A list of coffee items
+    list CoffeeItems {
+        member: CoffeeItem
+    }
+
+-------------------
+Defining Operations
+-------------------
+With the shapes defined above, let's create an operation on our service for returning a menu to the consumer:
+
+.. code-block:: smithy
+    :caption: ``main.smithy``
+
+    ...
+    service CoffeeShop {
+       version: "2024-04-04"
+       operations: [
+            GetMenu
+       ]
+    }
+
+    /// Retrieve the menu
+    @http(method: "GET", uri: "/menu")
+    @readonly
+    operation GetMenu {
+        output := {
+            items: CoffeeItems
+        }
+    }
+
+We've named the operation ``GetMenu``. It does not define an input, and models its output as a structure with a single
+member, ``items``, which contains ``CoffeeItems`` (a shape we defined above). With the ``restJson1`` protocol, a
```

Comment:
link to section above.

Reply:
Done, wish we could link to specific code block lines.